# ROS2 Slam Toolbox Configuration
slam_toolbox:
  ros__parameters:

    # ROS Parameters
    odom_frame: odom
    map_frame: map
    base_frame: base_footprint
    scan_topic: /scan
    mode: mapping # async mapping

    # Base Parameters
    debug_logging: false
    throttle_scans: 1
    transform_publish_period: 0.02 # 50Hz
    map_update_interval: 5.0
    resolution: 0.05 # 5cm per cell
    max_laser_range: 12.0 # for LD19
    minimum_time_interval: 0.5
    transform_timeout: 0.2
    tf_buffer_duration: 30.
    stack_size_to_use: 40000000 
    enable_interactive_mode: true

    # Scan Matching Parameters
    use_scan_matching: true
    use_scan_barycenter: true
    minimum_travel_distance: 0.1
    minimum_travel_heading: 0.1
    scan_buffer_size: 10
    scan_buffer_maximum_scan_distance: 10.
    link_match_maximum_distance: 1.5
    link_scan_maximum_distance: 1.5
    loop_search_maximum_distance: 3.0
    do_loop_closing: true
    loop_match_minimum_chain_size: 10
    loop_match_maximum_variance_co_variance: 0.5
    loop_match_minimum_response_co_variance: 1.0
    loop_match_threshold: 0.1
    loop_search_space_dimension: 8.0
    loop_search_space_resolution: 0.05
    loop_search_space_smear_deviation: 0.03

    # Scan Matcher
    distance_variance_penalty: 0.5      
    angle_variance_penalty: 0.5    

    fine_search_angle_offset: 0.00349     
    coarse_search_angle_offset: 0.349   
    coarse_angle_resolution: 0.0349        
    minimum_angle_penalty: 0.9
    minimum_distance_penalty: 0.5
    use_response_expansion: true
